lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -274,7 +274,7 @@ void LCIStrategicPlugin::handleFailureCase(const cav_srvs::PlanManeuversRequest&
 
   resp.new_plan.maneuvers.push_back(composeIntersectionTransitMessage(
       traffic_light_down_track, intersection_end_downtrack_.get(), intersection_speed_.get(),
-      incomplete_traj_params.modified_departure_speed, current_state.stamp + ros::Duration(incomplete_traj_params.modified_remaining_time), intersection_exit_time, crossed_lanelets.back().id(), exit_lanelet_id));
+      incomplete_traj_params.modified_departure_speed, current_state.stamp + ros::Duration(incomplete_traj_params.modified_remaining_time), intersection_exit_time, crossed_lanelets.back().id(), crossed_lanelets.back().id()));
```

Comment:
Can I get an explanation of this change? The exit_lanelet_id appears to be the actual final lanelet. The cross_lanelets.back() is likely the next lanelet since getLaneletsBetweenWithException was called with bounds_inclusive = true. This means the ending lanelet for the maneuver would actually be wrong.

Reply:
One discovery that was found during testing was that the IntersectionTransitManeuvering plugin only ever uses the first starting_lanelet_id in creating lanelet_ids for lane_follow maneuever:
https://github.com/usdot-fhwa-stol/carma-platform/blob/feature/uc3_full/intersection_transit_maneuvering/src/intersection_transit_maneuvering.cpp#L252

And the function getLaneletsBetweenWithException is called from current_downtrack to traffic_light_down_track which is the 'start of the intersection', so by calling it bounds inclusive will actually include the intersection lanelet as the last element of the list, which is why crossed_lanelets.back() is picked

much of the IntersectionTransitManeuver msg composing changes are related to that. Previously it worked because lanelet_ids field requirement in basic_autonomy was introduced later on